### PR TITLE
Update task dependencies

### LINKED ISSUE #166: Use task id to indicate dependencies in mission
**Describe the improvement you would like to see**
The current implementation of dependencies uses the index of the task in a mission to indicate dependencies. Instead the unique id of each task should be used to manage the dependencies.

**How will this change existing functionality?**
There will be requirements to find the correct task within the mission based on the id rather than by directly indexing the list.

**How will this improvement affect the current Threat Model?**
Not affected.

### PR DESCRIPTION
**Pull request by vetlek**
Closes #166

### CODE REVIEW

File: src/robot_interface/models/mission/task.py [hunk 1]
```diff
@@ -56,7 +56,9 @@ class InspectionTask(Task):
 
     inspections: List[Inspection] = field(default_factory=list, init=False)
 
-    pass
+    @staticmethod
+    def get_inspection_type():
+        return Inspection
```
Comment (aestene) [inline src/robot_interface/models/mission/task.py:61]:
```suggestion
    def get_inspection_type(self):
        return type(self)
```

File: src/isar/state_machine/state_machine.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from copy import deepcopy
 from typing import Deque, Optional, Tuple
 
+from black import re
```
Comment (aestene) [inline src/isar/state_machine/state_machine.py:7]: Should this be here?

**Review by aestene — CHANGES_REQUESTED** [review]
One minor comment for a line that might be removed, otherwise good.

File: src/isar/state_machine/state_machine.py (hunk 2 above)
Comment (vetlek) [reply src/isar/state_machine/state_machine.py:7]: Nope

**Review by aestene — APPROVED** [review]
LGTM

**Pull request merged**